config/internal/ml-metadata/metadata-envoy.deployment.yaml.tmpl

```diff
@@ -71,6 +71,7 @@ spec:
             - mountPath: /etc/envoy.yaml
               name: envoy-config
               subPath: envoy.yaml
+        {{ if .MLMD.Envoy.DeployRoute }}
```

Comment:
I'm not following why this is in an `if` block now. Can you explain?

Reply:
In previous tasks, we made the OAuth2 Proxy optional to deploy because KinD does not accept this sidecar. Just FYI we use KinD in our Integration Tests. You can see we do the same with API Server in https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/internal/apiserver/default/deployment.yaml.tmpl#L247

Reply:
got it. Did you need to add this as part of this PR to get tests working? (If not, I'd suggest it should be extracted to its own PR.)